tests/test_classical_gradients.py

```diff
@@ -367,12 +370,14 @@ def test_no_argnum_jacobian(self, tol):
 
         jac_fn = qml.jacobian(cost_fn)
         res = jac_fn(x, y)
-        expected = np.array([[np.cos(x) * np.cos(y), -np.sin(x) * np.sin(y)], [y ** 2, 2 * x * y]])
-        assert np.allclose(res, expected, atol=tol, rtol=0)
+        expected = (
+            np.array([np.cos(x) * np.cos(y), y ** 2]),
+            np.array([-np.sin(x) * np.sin(y), 2 * x * y]),
+        )
+        assert all(np.allclose(_r, _e, atol=tol, rtol=0) for _r, _e in zip(res, expected))
 
-        x = np.array(0.5, requires_grad=True)
-        y = np.array(0.2, requires_grad=False)
+        x = np.array(0.5, requires_grad=False)
+        y = np.array(0.2, requires_grad=True)
```

Comment:
What is the motivation for this change? :)

Reply:
I thought it might be a more delicate test like this, because there is one trainable argument to be inferred but it is not the first one. It's super minor though :)

Reply:
This makes a lot of sense actually 👍